### ISSUE TITLE
make on eve/build fails

### ISSUE THREAD

**Issue by code-recipes** [witheve/Eve#391]
Hi,
I'm trying to compile the latest alpha version you released and get the following message when doing a make on the eve/build path

`make: *** No rule to make target`compiler.o', needed by `eve'.  Stop.`

I'm using the latest `master` branch. My distro is Ubuntu 14.04 LTS with latest updates (kernel 3.13.0.91-generic)

**Comment by ibdknox** [comment]
if you run make from the root dir what happens?

**Comment by code-recipes** [comment]
I get this when running Eve from the root dir

`:~/Eve$ make`
`cd build ; make`
`make[1]: Entering directory /Eve/build'`
`make[1]: *** No rule to make target 'compiler.o', needed by 'eve'.  Stop.`
`make[1]: Leaving directory '/Eve/build'`
`make: *** [all] Error 2`

**Comment by ibdknox** [comment]
@convolvatron any thoughts?

**Comment by convolvatron** [comment]
sorry, didn't see this. i think its very likely that there isn't a luajit shaped exactly like
what its looking for. 

if you change the top line of build/Makefile from /usr/local to ./lua
it will attempt to build a private copy from the luajit repo

**Comment by code-recipes** [comment]
If this helps, this is the listing of my /usr/loca/bin dir

`:/usr/local/bin$ ls
bower               cg_annotate  luajit          pip     rustdoc    tsserver            vgdb
callgrind_annotate  cg_diff      ms_print        pip2    rust-gdb   valgrind
callgrind_control   cg_merge     multirust       pip2.7  rustup.sh  valgrind-di-server
cargo               composer     multirustproxy  rustc   tsc        valgrind-listener
:/usr/local/bin$ `

**Comment by dram** [comment]
Same problem here.

Eve compiled successfully after I adjust `LUAJIT_INSTALL` in `build/Makefile` to where LuaJIT installed.

**Comment by code-recipes** [comment]
@dram, thanks for the tip. Managed to get past the previous error, but now I get a different one. See below

`:~/Eve/build$ make`
`cc -std=gnu99 -I../csrc/unix -I../csrc/core -I../csrc -I. -I/usr/local/include/luajit-2.1 -g -c ../csrc/http/json.c`
`In file included from ../csrc/core/core.h:36:0,`
`                 from ../csrc/runtime.h:4,`
`                 from ../csrc/http/json.c:1:`
`../csrc/core/timer.h: In function ‘seconds’:`
`../csrc/core/timer.h:15:5: warning: left shift count >= width of type [enabled by default]`
`     return(((u64)n)<<32);`
`     ^`
`../csrc/core/timer.h: In function ‘milliseconds’:`
`../csrc/core/timer.h:20:5: warning: left shift count >= width of type [enabled by default]`
`     return((((u64)n)<<32)/1000ull);`
`     ^`
`../csrc/core/timer.h: In function ‘rdtsc’:`
`../csrc/core/timer.h:31:5: warning: left shift count >= width of type [enabled by default]`
`     return (((ticks)a) | (((ticks)d) << 32));`
`     ^`
`../csrc/http/json.c: In function ‘print_value_json’:`
`../csrc/http/json.c:7:5: warning: large integer implicitly truncated to unsigned type [-Woverflow]`
`     case uuid_space:`
`     ^`
`../csrc/http/json.c:10:5: warning: large integer implicitly truncated to unsigned type [-Woverflow]`
`     case float_space:`
`     ^`
`../csrc/http/json.c:10:5: error: duplicate case value`
`../csrc/http/json.c:7:5: error: previously used here`
`     case uuid_space:`
`     ^`
`../csrc/http/json.c:13:5: warning: large integer implicitly truncated to unsigned type [-Woverflow]`
`     case estring_space:`
`     ^`
`../csrc/http/json.c:13:5: error: duplicate case value`
`../csrc/http/json.c:7:5: error: previously used here`
`     case uuid_space:`
`     ^`
`make: *** [json.o] Error 1`

**Comment by margold** [comment]
Adjusting `$LUAJIT`, `$LIBLUAJIT`, and `$LUAJIT_INCLUDES` at the beginning of `build/Makefile` worked for me. (I have luajit installed with homebrew on OSX; with eve at commit `7060081c0360`)

**Comment by convolvatron** [comment]
@code-recipes what platform are you compiling on? this is almost certainly due to 
using a 32 bit environment. this is totally fixable, but i'll have to bring one up to work
through the issues. can you confirm thats the case?

**Comment by convolvatron** [comment]
@code-recipes perfect, i'll try to set up and environment and commit a patch

**Comment by convolvatron** [comment]
@code-recipes i checked in a clean 32 bit build on clang into master. it'll take me a bit longer
to actually test it on 32 bits, but if you feel like giving it a whirl

**Comment by convolvatron** [comment]
oh, my. i installed a 32 bit vm, installed the tools, went through the process of building eve, and
realized far too late that the way the system is structured, it really does need a 64 bit address space. while this is convenient it isn't strictly necessary.

so unfortunately its going to take a bit more investment of time before i can put in shims so that it will run on a 32 bit machine.

if you're curious, we used a very old technique where instead of adding type tags to objects, we used  the virtual memory system to encode the type directly in the pointer to the object. so, different types live in different parts of the address space. this exploits the fact that 64 bit address have so much room that we can reserve bits for types and have plenty of room left over. 

a 32 bit space (really closer to 30 after the runtime and the loader are finished) doesn't really tolerate such a profligate waste, so i have some slightly slower support I can add for dynamic type tags optionally compiled in...but it won't be until next week that i will have time to bring it up and test it on both platforms
